src/components/inputs/multi-select/multi-select.component.tsx

```diff
@@ -16,44 +16,39 @@ import styles from './multi-select.scss';
 const MultiSelect: React.FC<FormFieldProps> = ({ question, onChange, handler, previousValue }) => {
   const { t } = useTranslation();
   const [field, meta] = useField(question.id);
-  const { setFieldValue, encounterContext, layoutType, workspaceLayout } = React.useContext(FormContext);
+  const { setFieldValue, encounterContext, layoutType, workspaceLayout, isFieldInitializationComplete } =
+    React.useContext(FormContext);
   const [errors, setErrors] = useState([]);
   const [warnings, setWarnings] = useState([]);
   const [counter, setCounter] = useState(0);
-  const [touched, setTouched] = useState(false);
   const isFieldRequiredError = useMemo(() => errors[0]?.errCode == fieldRequiredErrCode, [errors]);
 
-  useEffect(() => {
-    // Carbon's MultiSelect has issues related to not updating the component based on the `initialSelectedItems` prop
-    // this is an intermittent issue. As a temporary solution, were are forcing the component to re-render
-    if (field.value && field.value.length == 0) {
-      // chances are high the value was cleared
-      // force the Multiselect component to be re-mounted
-      setCounter(counter + 1);
-    } else if (!touched && question.value) {
-      setCounter(counter + 1);
-    }
-  }, [field.value]);
-
   useEffect(() => {
     if (question['submission']) {
       question['submission'].errors && setErrors(question['submission'].errors);
       question['submission'].warnings && setWarnings(question['submission'].warnings);
     }
   }, [question['submission']]);
 
+  const selectOptions = question.questionOptions.answers
+    .filter((answer) => !answer.isHidden)
+    .map((answer, index) => ({
+      id: `${question.id}-${answer.concept}`,
+      concept: answer.concept,
+      label: answer.label,
+      key: index,
+      disabled: answer.disable?.isDisabled,
+    }));
```

Comment:
Can we memoise this?

Reply:
In my first approach memoised it but for some reason (will be addressed during the formik migration), some state events weren't properly being propagated. An example is the `answer.disable. isDisabled`. This property is computable based on skip logic, when I have it like:

```ts
const selectOptions = useMemo(
    () =>
      question.questionOptions.answers
        .filter((answer) => !answer.isHidden)
        .map((answer, index) => ({
          id: `${question.id}-${answer.concept}`,
          concept: answer.concept,
          label: answer.label,
          key: index,
          disabled: answer.disable?.isDisabled,
        })),
    [question],
  );

```

The new state computation of `answer.disable?.isDisabled` doesn't get propagated after skip logic runs. I guess it's the same reason why it wasn't [memoised originally](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/components/inputs/multi-select/multi-select.component.tsx#L99-L107).